examples/03_connectivity/plot_seed_based_correlation.py

```diff
@@ -0,0 +1,151 @@
+# Exemplarily, we can also select 5 random voxels from the **brain-wide data** and plot the time series from.
+
+plt.plot(brain_time_series[:, [10, 45, 100, 5000, 10000]])
+plt.title('Time series from 5 random voxels')
+plt.xlabel('Scan number')
+plt.ylabel('Normalized signal')
+plt.tight_layout()
+
+
+
+################################################
+# Performing the seed-based correlation analysis
+# ----------------------------------------------
+#
+# Now that we have two arrays (**sphere signal**: (n_volumes, 1), **brain-wide voxel-wise signal** (n_volumes, n_voxels)), we
+# can correlate the **seed signal** with the **signal of each voxel**. The dot product of the two arrays will give us this
+# correlation. Note that the signals have been variance-standardized during extraction. To have them standardized to
+# norm unit, we further have to divide the result by the length of the time series.
+import numpy as np
+
+seed_based_correlations = np.dot(brain_time_series.T, seed_time_series) / seed_time_series.shape[0]
+
+################################################
+# The resulting array will contain a value representing the correlation values between the signal in the **seed region** of
+# interest and **each voxel's signal**, and will be of shape (n_voxels, 1). The correlation values can potentially range
+# between -1 and 1.
+print("seed-based correlation shape: (%s, %s)" % seed_based_correlations.shape)
+print("seed-based correlation: min = %.3f; max = %.3f" % (seed_based_correlations.min(), seed_based_correlations.max()))
+
+
+################################################
+# Fisher-z transformation and save nifti
+# ----------------------------------------------
+# Now we can Fisher-z transform the data to achieve a normal distribution. The transformed array can now have values
+# more extreme than +/- 1.
+seed_based_correlations_fisher_z = np.arctanh(seed_based_correlations)
+print("seed-based correlation Fisher-z transformed: min = %.3f; max = %.3f" % (
+    seed_based_correlations_fisher_z.min(), seed_based_correlations_fisher_z.max()))
+
+# Finally, we can tranform the correlation array back to a Nifti image object, that we can save.
+seed_based_correlation_img = brain_masker.inverse_transform(seed_based_correlations.T)
+seed_based_correlation_img.to_filename('sbc_z.nii.gz')
+
+
+################################################
+# Plotting the seed-based correlation map
+# ----------------------------------------------
+# We can also plot this image and perform thresholding to only show values more extreme than +/- 0.3.
+# Furthermore, we can display the location of the seed with a sphere and set the cross to the center
+# of the seed region of interest.
+from nilearn import plotting
+
+display = plotting.plot_stat_map(seed_based_correlation_img, threshold=0.3, cut_coords=dmn_coords[0])
+display.add_markers(marker_coords=np.array(dmn_coords), marker_color='g', marker_size=300)
```

Comment:
It fails without np.array

  File "<string>", line 6, in <module>
  File "/Users/franzliem/Dropbox/Workspace/nilearn/nilearn/plotting/displays.py", line 740, in add_markers
    marker_coords, direction, return_direction=True)
  File "/Users/franzliem/Dropbox/Workspace/nilearn/nilearn/plotting/displays.py", line 209, in _coords_3d_to_2d
    return coords_3d[:, dimensions], coords_3d[:, index]
TypeError: list indices must be integers, not tuple

> Am 26.02.2016 um 16:27 schrieb Gael Varoquaux <notifications@github.com>:
> 
> In examples/03_connectivity/plot_seed_based_correlation.py <https://github.com/nilearn/nilearn/pull/1024#discussion_r54256888>:
> 
> > +
> > +# Finally, we can tranform the correlation array back to a Nifti image object, that we can save.
> > +seed_based_correlation_img = brain_masker.inverse_transform(seed_based_correlations.T)
> > +seed_based_correlation_img.to_filename('sbc_z.nii.gz')
> > +
> > +
> > +################################################
> > +# Plotting the seed-based correlation map
> > +# ----------------------------------------------
> > +# We can also plot this image and perform thresholding to only show values more extreme than +/- 0.3.
> > +# Furthermore, we can display the location of the seed with a sphere and set the cross to the center
> > +# of the seed region of interest.
> > +from nilearn import plotting
> > +
> > +display = plotting.plot_stat_map(seed_based_correlation_img, threshold=0.3, cut_coords=dmn_coords[0])
> > +display.add_markers(marker_coords=np.array(dmn_coords), marker_color='g', marker_size=300)
> Do you need the "np.array" here? If so, we need to fix add_markers.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/nilearn/nilearn/pull/1024/files#r54256888>.
>